server/webhook_notifier.go

```diff
@@ -0,0 +1,157 @@
+package server
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"log"
+	"net"
+	"net/http"
+	"time"
+)
+
+// WebhookNotifier implements ConnectionNotifier by sending a POST request to a webhook URL.
+// The payload is a JSON object with the following fields:
+//
+//	event: the event type, one of "connecting", "missing-backend", "failed-backend-connection", "success"
+//	status: the status of the event, one of "missing-backend", "failed-backend-connection", "success"
+//	client-host: the client host
+//	client-port: the client port
+//	server-address: the server address
+//	player-info: the player info, if any
+//	backend: the backend host and port, if any
+//	error: the error, if any
+type WebhookNotifier struct {
+	url         string
+	requireUser bool
+
+	client *http.Client
+}
+
+const (
+	WebhookEventConnecting = "connecting"
+)
+
+const (
+	WebhookStatusMissingBackend          = "missing-backend"
+	WebhookStatusFailedBackendConnection = "failed-backend-connection"
+	WebhookStatusSuccess                 = "success"
+)
+
+type WebhookNotifierPayload struct {
+	Event           string      `json:"event"`
+	Status          string      `json:"status"`
+	ClientHost      string      `json:"client-host"`
```

Comment:
I would include client host and port under player info as mentioned earlier. If player info like name or uuid is unavailable, I would just set it to nullish values (empty string: "", uuid of 0's)

Reply:
The originating client IP+port is always available even during a server list/ping operation and refers to the client machine. Player info is separately provided to allow for webhook consumer to differentiate a ping from a user attempting a session.

Reply:
> The originating client IP+port is always available even during a server list/ping operation and refers to the client machine. Player info is separately provided to allow for webhook consumer to differentiate a ping from a user attempting a session.

If that is the case then, it would probably be better to include under a "client" object. I believe Javascript wouldn't be able to reference this in object notation and would instead have to handle it as `const host = body["client-host"]`

![image](https://github.com/user-attachments/assets/fe7405b0-0614-4bad-9ee6-9022c96a20de)

Reply:
Moved into a `client` object.